### PR TITLE
made small changes to login screen

### PR DESCRIPTION
**Pull request by rpirritano**
Made all the recommended changes to the login screen, as well as having wrong password message no longer appear with clicking forgot password link

### CODE REVIEW

File: .htaccess [hunk 1]
```diff
@@ -78,3 +78,7 @@ Options -Indexes
 <IfModule pagespeed_module>
   ModPagespeed Off
 </IfModule>
+#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####
```
Comment (rullzer) [inline .htaccess]: Don't commit to .htaccess please ;)

**Review by rullzer — CHANGES_REQUESTED** [review]

**Comment by jancborchardt** [comment]
@rpirritano thanks for the contribution! :) Could you clarify what you mean by recommended changes, ideally with a before/after screenshot for easier testing?

cc @nextcloud/designers

**Comment by rpirritano** [comment]
@jancborchardt , sure!  The small changes I made were recommended from this thread: https://github.com/nextcloud/server/issues/7265#issuecomment-348094139

Below I am copying the recommendations, and before/after screenshots:

1. Logo is too high up; actually, its distance from top is OK, but now it's too far from the form itself:
image

I would remove 16px top+bottom margins from form {} in guest.css since they don't seem

Before: 

before1

After:

screen shot 2018-01-06 at 2 01 10 pm

2. Is there a reason why footer has height: 70px? We can easily remove it (I think) and change the p.info {margin: 0 auto; padding-top: 20px; to simpler p.info {margin: 20px auto;} Added advantage to these changes would mean that the whole page would fit nicely on 320*480 screen:

Before: 

before2

After:

screen shot 2018-01-06 at 2 01 10 pm

3. "Wrong password " warning is above the #submit# button, but the#lost-password` warning is bellow - makes no sense at all
3a. Additionally, "Lost Password" warning also has opacity: .7; added to it for some reason.
3b. Ideally, it should also disappear as soon as "Forgot password?" button is clicked. That way, only one error should be visible at any given moment.

Before:
before3

After:

screen shot 2018-01-06 at 2 00 36 pm

screen shot 2018-01-06 at 2 00 55 pm

**Comment by skjnldsv** [comment]
> would fit nicely on 320*480 screen

Is there really many out there tough? :)
I think it's great to have a small bottom padding to the slogan, We don't want the footer to be just at the bottom ;)

**Comment by jancborchardt** [comment]
@rpirritano seems mostly good – however you seem to have removed the border from the primary buttons? Let's not do that as it makes them not look as clickable, which is a big problem for the primary action. :)

**Comment by rpirritano** [comment]
@jancborchardt , this pull request does have the correct code and does have the button borders.  Those screenshots above were accidentally taken from an older version.

I double checked my code by cloning this master, and it appears correct.  Sorry for the confusion.

**Comment by jancborchardt** [comment]
@rpirritano could you update the screenshots? :)

@nextcloud/designers what do you think?

**Comment by rpirritano** [comment]
ok updated the comment above with the correct "after" screenshots

**Comment by skjnldsv** [comment]
- [ ] The footer height needs to stay at 70px
- [ ] The .htaccess file needs to be removed from this pr

**Comment by enoch85** [comment]
:+1: Design wise from me. Looks a lot better.

File: .htaccess [hunk 2]
```diff
@@ -1,80 +0,0 @@
...
-  AddEncoding gzip svgz
-</IfModule>
-<IfModule mod_dir.c>
-  DirectoryIndex index.php index.html
-</IfModule>
-AddDefaultCharset utf-8
-Options -Indexes
-<IfModule pagespeed_module>
-  ModPagespeed Off
-</IfModule>
```
Comment (MorrisJobke) [inline .htaccess]: Now you have removed the whole .htaccess 😉 I guess there was a misunderstanding: it was only regarding the changes in .htaccess. Could you try to do that? Thanks a lot 👍

Comment (rpirritano) [reply .htaccess]: Just to clarify, your comment above, "Now you have removed the whole .htaccess "  I did remove that file?  Was I only supposed to remove changes in the file and if so, what changes?

Comment (MorrisJobke) [reply .htaccess]: Those changes from your first commit: https://github.com/nextcloud/server/pull/7650/commits/409c2a08f8927c41340ba1fba14f76c815ed8f42#diff-8052c42ab3b8aa06a3f5f788a4ddccc2R84

File: .htaccess (hunk 1 above)
Comment (MorrisJobke) [reply .htaccess]: @rpirritano Now the changes are back again :)

Comment (rpirritano) [reply .htaccess]: sorry I am new to this and the apache server.  I removed the changes that you took a screen shot of.  Can you also do that for the changes that are back again?

**Comment by rpirritano** [comment]
ok I deleted the changes and hope this worked.

**Comment by MorrisJobke** [comment]
> ok I deleted the changes and hope this worked.

Don't worry. Sadly it didn't worked out. Best would be to squash all 5 commits into 1 commit by rebasing the branch, but this is not that trivial. I would give you a step by step guide, but sadly I need to leave now. I will provide you the guide later (or maybe @rullzer or @nickvergessen can help you in the meantime).

Thanks for the contribution anyways.

**Comment by rpirritano** [comment]
A guide would be good, in the meantime I will research it myself and give it a go.

**Comment by MorrisJobke** [comment]
Let me do the guide:

First do a backup of the current situation in case something goes wrong:

```
$ git branch master-backup
```

Now we do an interactive rebase (which allows to squash all commits into one) based on the last commit from the Nextcloud master branch (`08cbfb6c6b554eed221aafe185c02dc2e7841d95` is the commit before the first commit in this PR :wink:):

```
$ git rebase -i 08cbfb6c6b554eed221aafe185c02dc2e7841d95
```

An editor should open, that looks like this:

bildschirmfoto 2018-01-10 um 21 28 56

This shows you the 4 commits of your master branch and you can specify an operation per commit. Leave the `pick` on the first line and replace the `pick` with an `s` (like squash - see comment in the editor) in the 3 other commits. This will keep the first commit and squash the 3 others into this commit to generate a big commit. (Keep in mind to have a space between the `s` and the commit sha).

Then another editor opens, that will allow you to specify the commit message for this new commit. Adjust it like you want.

Now copy the .htaccess file from our current master (this is easier than to fiddle around with partial git reverts):

```
wget https://raw.githubusercontent.com/nextcloud/server/master/.htaccess -O .htaccess
```

Just add this new file to the staged area in git and amend the previous commit to get rid of this change (amending adds the current change to the last commit without creating a new one):

```
git add .htaccess
git commit --amend
```

Now you are ready for pushing your changes, but because you changed the git history you need to do a force push to overwrite the git history on your master branch:

```
git push -f origin master
```

You are done 🚀

**Comment by rpirritano** [comment]
Thanks, the guide was easy to follow!

**Review by rullzer — APPROVED** [review]

**Comment by MorrisJobke** [comment]
Let me check why the integration tests fail.

**Comment by MorrisJobke** [comment]
@rpirritano Could you add following patch:

```patch
diff --git a/tests/acceptance/features/bootstrap/LoginPageContext.php b/tests/acceptance/features/bootstrap/LoginPageContext.php
index 560dd83f30..1496e3030c 100644
--- a/tests/acceptance/features/bootstrap/LoginPageContext.php
+++ b/tests/acceptance/features/bootstrap/LoginPageContext.php
@@ -66,7 +66,7 @@ class LoginPageContext implements Context, ActorAwareInterface {
 	 * @return Locator
 	 */
 	public static function wrongPasswordMessage() {
-		return Locator::forThe()->xpath("//*[@class = 'warning' and normalize-space() = 'Wrong password.']")->
+		return Locator::forThe()->xpath("//*[@class = 'warning wrongPasswordMsg' and normalize-space() = 'Wrong password.']")->
 				describedAs("Wrong password message in Login page");
 	}
 ```

Once you changed the file do this:

```
git add tests/acceptance/features/bootstrap/LoginPageContext.php
git commit --amend
git push -f origin master
```

Then the tests should pass

**Comment by MorrisJobke** [comment]
Problem here: As you changed the class of the warning the selector was not working anymore.

**Comment by rullzer** [comment]
I took the liberty to push that so we can merge it.
Again, good stuff @rpirritano  :D

**Review by juliusknorr — APPROVED** [review]
Fine by me, if the tests pass. :+1:

**Comment by rpirritano** [comment]
since this was merged, do I still need to add that patch @MorrisJobke

**Comment by juliushaertl** [comment]
@rpirritano That patch was already added by @rullzer, so everything is done here. 
Thanks for the contribution, further pull requests are always welcome. :wink:

**Comment by MorrisJobke** [comment]
> @rpirritano That patch was already added by @rullzer, so everything is done here.
> Thanks for the contribution, further pull requests are always welcome. 😉

Correct :) 


@rpirritano Thanks for you contribution here :) You could join our chat on Freenode - the channel name is #netcloud-dev. If you are looking for other issues, that one labeled with "starter issues" are maybe a good entry point for other things to do: https://github.com/nextcloud/server/issues?q=is%3Aissue+is%3Aopen+label%3A%22starter+issue%22

**Comment by rpirritano** [comment]
Great! Thank you for all your help!

**Pull request merged**